File: frontend/src/modules/auth/validators/NameValidator.ts
/**
 * NameValidator — Chain of Responsibility node for name presence (FE).
 * Extends BaseValidator (core). SOLID: SRP·OCP·LSP.
 */

import { BaseValidator, type ValidationResult } from '../../../core/base/BaseValidator'

export class NameValidator extends BaseValidator {
  validate(data: unknown): ValidationResult {
    const nombre = String((data as { nombre?: string })?.nombre ?? '').trim()
    if (!nombre) {
      return { isValid: false, errors: ['El nombre es obligatorio.'], field: 'nombre' }
    }
    return { isValid: true, errors: [], field: 'nombre' }
  }
}

File: backend/apps/authentication/validators/name_validator.py
"""
NameValidator — Chain of Responsibility node validating name presence (SRP).
Extends BaseValidator (S1). Pattern: Chain of Responsibility. SOLID: SRP·OCP·LSP.
"""

from __future__ import annotations

from core.base.base_validator import BaseValidator, ValidationResult


class NameValidator(BaseValidator):
    def validate(self, data: dict) -> ValidationResult:
        nombre = (data.get("nombre") or "").strip()
        if not nombre:
            return ValidationResult(
                is_valid=False,
                errors=["El nombre es obligatorio."],
                field_name="nombre",
            )
        return ValidationResult(is_valid=True)
